tools/shared.py

```diff
@@ -772,6 +772,32 @@ def clean_temp():
     prepare_to_clean_temp(EMSCRIPTEN_TEMP_DIR) # this global var might change later
   return EMSCRIPTEN_TEMP_DIR
 
+class WarningManager:
```

Comment:
I think it might make more sense to have `WarningManager` just implement a `warn()` method. Over time we could convert all the emscripten-specific warnings to use it.
There would be 2 advantages to that. First, you wouldn't have to add yet-another case to the if conditions that emit the warning. But even better, you could then support flags like `-W` (disable all warnings) and other centralized ways of controlling warnings.
You'd sometimes still have to pass it the type of warning in some cases, so it could be something like this:

```
class WarningManager:
  def warn(type, text):
    if <type is not suppressed>:
      logging.warning(text)
```
and in the emcc code,
```
shared.WarningMananger.warn(shared.WarningManager.WARN_SEPARATE_ASM, 'warning!')
```
Or maybe even get rid of the enums and just use a string that matches the warning name in the flag, like 'absolute-paths'. That might make the `warn` code cleaner (see my suggestion below).
Not sure which of those is better though, I still generally like enums more than strings. Hmm...